### ISSUE TITLE
[PRE REVIEW]: PyLUR: A land use regression wizard for ArcGIS Pro

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1753]
**Submitting author:** @anmolter (<a href="http://orcid.org/0000-0002-0824-4478">Anna Mölter</a>)
**Repository:** <a href="https://github.com/anmolter/PyLUR" target ="_blank">https://github.com/anmolter/PyLUR</a>
**Version:** v2.2.4 or higher
**Editor:** @hugoledoux
**Reviewers:** @Athelena, @fortinma

**Author instructions**

Thanks for submitting your paper to JOSS @anmolter. **Currently, there isn't an JOSS editor assigned** to your paper.

@anmolter if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```


**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1753 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-efe915e61673/lib/whedon.rb:135:in `check_fields': Paper YAML header is missing expected fields: affiliations (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-efe915e61673/lib/whedon.rb:87:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-efe915e61673/lib/whedon/processor.rb:36:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-efe915e61673/lib/whedon/processor.rb:36:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-efe915e61673/bin/whedon:55:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-efe915e61673/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by kthyng** [comment]
Hi @anmolter, I have some concerns about your submission that need to be addressed before we can move forward on your submission. As it stands, it is failing on the [submission requirement](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements) of 

> Be stored in a repository that is browsable online without registration

since to look at the files, one needs to download a big zip file and unzip. Therefore, the packaging of your submission needs improvement before we can review it.

Other important improvements include documentation and tests for your code.

**Comment by anmolter** [comment]
I have removed the zip files and replaced them with browsable folders in the repository. I have changed the README.md, so it now recommends to clone or download the repository and it shows that the documentation and tutorial is in the manual. 
I have also fixed the typo in paper.md, so it is now 'affiliations' instead of 'affiliation'.

**Comment by anmolter** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1753 with the following error: 

 Could not find bibliography file: PyLUR_paper.bib
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by anmolter** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1753 with the following error: 

 Error reading bibliography ./paper.bib (line 14, column 13):
unexpected "H"
expecting space or ","
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by anmolter** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01753/joss.01753/10.21105.joss.01753.pdf)

**Comment by danielskatz** [comment]
I think we can start the process of editing and review, but I expect there will be a fairly large number of issues brought up by the reviewers, as this code doesn't appear to be as modular as I would expect, and there are machine dependencies, including the fact that the instructions to read the manual likely won't work on a non-PC.

**Comment by danielskatz** [comment]
👋 @leouieda - I know you have a fair number of papers now, but would you also be willing to take on this one?

**Comment by danielskatz** [comment]
👋 @leouieda - I know you have a fair number of papers now, but would you also be willing to take on this one?

**Comment by leouieda** [comment]
Hi @danielskatz I'd be happy edit this submission :+1:

**Comment by leouieda** [comment]
@whedon assign @leouieda as editor

**Comment by whedon** [comment]
OK, the editor is @leouieda

**Comment by leouieda** [comment]
@anmolter I'm currently searching for reviewers for your submission and having a look at your paper and package. Since ArcGIS is required to test the software, I don't know how hard it will be find a willing reviewer who has a license. So this might take a little longer than usual.

While I'm searching for reviewers, please have a look at the [JOSS review criteria](https://joss.readthedocs.io/en/latest/review_criteria.html). This is what reviewers will be looking for in your software and our goal is to help you meet all of these criteria. If you want to get a head start in the review, I would suggest trying to work on some of the things that are missing in PyLUR (in particular, the [documentation part](https://joss.readthedocs.io/en/latest/review_criteria.html#documentation) since that is what reviewers are likely to check first).

Let me know if you have any questions and thank you for your patience :pray:

**Comment by leouieda** [comment]
:wave: @benbovy @zsylvester would you be interested in reviewing this submission for JOSS? You can find the [paper here](https://github.com/openjournals/joss-papers/blob/joss.01753/joss.01753/10.21105.joss.01753.pdf) and the [software repository here](https://github.com/anmolter/PyLUR). Please note that the software requires ArcGIS and you would need access to a license in order to test it. Thank you for your time.

**Comment by benbovy** [comment]
Sorry @leouieda I'm afraid I won't be of great help here, I don't have access to a ArcGIS license (and I'm not working on a Windows PC).

**Comment by anmolter** [comment]
@leouieda all the documentation is in the PyLUR_Manual, which is in the PyLUR folder (as explained in the README file). The manual is an html file, which has been rendered from a number of smaller html files via R markdown. Do you want me to add a copy of it to the top level of GitHub and to rename it 'documentation'?
Also, I'll be on maternity leave from tomorrow until the beginning of May 2020. I'll be checking my emails occassionally, but I may take a while to respond to any queries or actions.

**Comment by leouieda** [comment]
@benbovy thanks for the quick response :+1:

**Comment by leouieda** [comment]
> Do you want me to add a copy of it to the top level of GitHub and to rename it 'documentation'?

@anmolter it might help people find the actual files. Having `docs` or `documentation` is standard practice in the top level of the repository seems to be standard practice. It might also help if you place the HTML files in a `gh-pages` branch so it can be served by [Github Pages](https://help.github.com/en/categories/working-with-github-pages). This would address the concerns @danielskatz had about the `.bat` file to open the HTML. 

> Also, I'll be on maternity leave from tomorrow until the beginning of May 2020. I'll be checking my emails occassionally, but I may take a while to respond to any queries or actions.

Congratulations! :confetti_ball: Just so you know, we have the option of pausing the review until you're back from your leave. Unlike most journals where you get a report from reviewers and then work on it, JOSS reviews tend to be more of a back-and-forth between reviewers and authors working together to improve the submission. It's no problem at all if you don't want to have to worry about responding to reviewer questions and suggestions. It's also not a problem if you want to continue with the review. Let me know what you prefer.

**Comment by anmolter** [comment]
@leouieda I've added a pdf of the user manual to my GitHub pages and I've updated the Readme file to include a link to the pdf. I hope that is ok now? 
With regards to maternity leave, could we put the review on hold until February 2020? Hopefully, things will be less crazy by then and I will be able to address reviewers' comments. If things are still crazy by then and I need a longer break, I'll message you to let you know.

**Comment by leouieda** [comment]
@anmolter I'll pause the review process until February 2020 and wait until closer to that date to actually assign reviewers. Let me know by commenting on this thread if you wish to resume the review earlier or later. Thank you and enjoy your leave!

**Comment by leouieda** [comment]
@openjournals/joss-eics could I mark this review as "paused"? The sole author is going on maternity leave until Feb 2020 and won't be available to respond to reviewers. I can ask whedon set a reminder for that date.

**Comment by labarba** [comment]
Sure. I just added the label. You can go ahead and set the reminder.

**Comment by zsylvester** [comment]
Hi @leouieda -- I don't use ArcGIS so probably I am not the right person to review it.

**Comment by whedon** [comment]
Sorry, I can't set reminders on PRE-REVIEW issues.

**Comment by leouieda** [comment]
> Sorry, I can't set reminders on PRE-REVIEW issues.

Well, that's inconvenient :man_shrugging:. @anmolter could you please post a message here when you're ready to come back and restart the review process? I'll put a reminder on my calendar as well to check in.

**Comment by leouieda** [comment]
@zsylvester thank you for considering!

**Comment by anmolter** [comment]
@leouieda Could you restart the review process please?

**Comment by leouieda** [comment]
Hi @anmolter welcome back! I have removed the paused label and will restart consideration of your submission.

**Comment by leouieda** [comment]
One thing that would help is if you could suggest some reviewers. Please include their names or github handles **without** using `@username` to they don't get a notification.

**Comment by anmolter** [comment]
Thanks @leouieda. Looking at the joss reviewers list martibosch or martinfleis could be suitable?

**Comment by anmolter** [comment]
Hi @leouieda, any luck with finding a reviewer?

**Comment by leouieda** [comment]
@anmolter sorry for the delay. I had messaged a couple of people who couldn't review.

**Comment by leouieda** [comment]
@martinfleis @martibosch would you be interested in reviewing this submission for JOSS? You can find the paper here and the software repository at the top of this issue. Please note that the software requires ArcGIS and you would need access to a license in order to test it. Thank you for your time.

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01753/joss.01753/10.21105.joss.01753.pdf)

**Comment by martinfleis** [comment]
@leouieda I don not have an access to ArcGIS Pro. Sorry.

**Comment by martibosch** [comment]
@leouieda although I think I could get access to ArcGIS through an EPFL student license or the like, I have never used and therefore I believe that other reviewers would be more suitable than me. Nonetheless, feel free to ping me back if you no other reviewers are available. Best,
Martí

**Comment by leouieda** [comment]
Thank you for the quick responses! I'll continue my search.

**Comment by anmolter** [comment]
@leouieda Do you want me to get some more names from the JOSS reviewers list? Or some people that are not on the reviewer list?

**Comment by leouieda** [comment]
@anmolter that would be great. I got back some "maybe"s in my search. The main challenge is finding people with access to ArcGIS Pro. Can PyLUR run on the non-Pro version as well? That might simplify things.

**Comment by anmolter** [comment]
Hi @leouieda  the following are on the JOSS reviewer list and use GIS and Python: jsta , hugoledoux , sgrieve , Scivision , kbevers , juliohm , sjsrey , hertzsprung , nagellette. Outside the JOSS reviewers list OliverSchmitz , zross , rsbivand probably have the right expertise. 

I did a version of PyLUR for ArcGIS Desktop for our project partners, but some of the code is different to the version I've uploaded here and installing additional Python packages in ArcDesktop can be a bit of a pain. Plus it is quite slow in ArcDesktop and can run out of memory with large files, so I wouldn't really want to upload the ArcDesktop version.

**Comment by leouieda** [comment]
@anmolter thank you for the recommendations and clarification. I'll keep trying.

**Comment by leouieda** [comment]
:wave: @hugoledoux @sgrieve would you be interested in reviewing this submission for JOSS? You can find the paper here and the software repository at the top of this issue. Please note that the software requires ArcGIS Pro and you would need access to a license in order to test it. Thank you for your time.

**Comment by hugoledoux** [comment]
I don't have access to ArcGIS Pro.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by arfon** [comment]
:wave: just a friendly check-in to see how you're getting on finding reviewers here @leouieda?

**Comment by hugoledoux** [comment]
@leouieda @arfon I have a colleague at TUDelft who is willing to do it. 

She is Athelena on GitHub

**Comment by leouieda** [comment]
Hi @arfon and @hugoledoux thank you reminder and finding a willing reviewer! So far I had gotten a few maybes but no commitments. The limiting factor seems to be access to ArcGIS Pro. My guess is that the overlap between people with a license and those doing Python and OSS is not very large. 

I'm hesitant to proceed to review with only a single reviewer, though. Particularly since I don't have access to ArcGIS Pro myself as a backup.

**Comment by hugoledoux** [comment]
Would that help if I take this one over? I think I can find people, in my "crowd" there are ArcGIS I reckon, pretty sure I can find another suitable one.

**Comment by leouieda** [comment]
@hugoledoux thank you for offering! It might be best since I'm poorly connected in the GIS world. I really appreciate it and hope you can get this moving forward. Let me know if I can help in any way.

**Comment by hugoledoux** [comment]
@leouieda sure I'll do that, happy to help.

@arfon not sure if that requires a formal change in the system but that would help me keep an overview.

**Comment by arfon** [comment]
> @arfon not sure if that requires a formal change in the system but that would help me keep an overview.

Sure thing. I can make the change here (you both also have the permissions to do this too 👇)

**Comment by arfon** [comment]
@whedon assign @hugoledoux as editor

**Comment by whedon** [comment]
OK, the editor is @hugoledoux

**Comment by hugoledoux** [comment]
Hello @Athelena & @fortinma, I'm happy that you have accepted to do this review.

I know it's your first time, so don't hesitate to ask question. You are also allowed to interact with the author (and me) here directly; the whole process is fully open to everyone. 

(It's also my first review to handle, so I guess I have to figure things out too. "whedon" is our bot that helps us automate the review process.)

Enjoy!

**Comment by hugoledoux** [comment]
@whedon assign @Athelena as reviewer

**Comment by whedon** [comment]
OK, @Athelena is now a reviewer

**Comment by hugoledoux** [comment]
@whedon add @fortinma as reviewer

**Comment by whedon** [comment]
OK, @fortinma is now a reviewer

**Comment by hugoledoux** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/2177.

**Comment by fortinma** [comment]
@hugoledoux , really sorry to ask but I am unable to edit the checklist. I fear i may not have accepted the invitation that has now expired. I thought i had. My apologies for the trouble.

**Comment by arfon** [comment]
> @hugoledoux , really sorry to ask but I am unable to edit the checklist. I fear i may not have accepted the invitation that has now expired. I thought i had. My apologies for the trouble.

@fortinma - I just re-invited you in #2177

**Comment by fortinma** [comment]
@anmolter I am having a hard time getting past the installation. I keep getting a pandas error. see dialog box screen cap attached.
<img width="443" alt="Screen Shot 2020-04-28 at 9 39 49 AM" src="https://user-images.githubusercontent.com/1124016/80505307-27fd9b00-8942-11ea-842c-510c21cbba14.png">
